src/Compilers/CSharp/Portable/Symbols/NullableAnnotationExtensions.cs

```diff
@@ -52,5 +53,28 @@ internal static CodeAnalysis.NullableAnnotation ToPublicAnnotation(this Nullable
             Debug.Assert((CodeAnalysis.NullableAnnotation)(NullableAnnotation.Oblivious + 1) == CodeAnalysis.NullableAnnotation.Disabled);
             return (CodeAnalysis.NullableAnnotation)annotation + 1;
         }
+
+        internal static NullabilityInfo ToNullabilityInfo(this CodeAnalysis.NullableAnnotation annotation, TypeSymbol type)
+        {
+            if (annotation == CodeAnalysis.NullableAnnotation.NotApplicable)
+            {
+                return default;
+            }
+
+            NullableAnnotation internalAnnotation = (NullableAnnotation)annotation - 1;
+            return internalAnnotation.ToNullabilityInfo(type);
+        }
+
+        internal static NullabilityInfo ToNullabilityInfo(this NullableAnnotation annotation, TypeSymbol type)
+        {
+            var flowState = TypeWithAnnotations.Create(type, annotation).ToTypeWithState().State;
+            return new NullabilityInfo(annotation.ToPublicAnnotation(), flowState.ToPublicFlowState());
+        }
+
+        internal static NullableAnnotation ToInternalAnnotation(this CodeAnalysis.NullableAnnotation annotation)
```

Comment:
>ToInternalAnnotation [](start = 43, length = 20)

Is this method necessary? (Do we need to convert from the public enum to the internal enum?)

Reply:
The symbol display API needs it currently. Hopefully after refactoring, it'll convert internal values to public and we won't need this.

---
In reply to: [269208962](https://github.com/dotnet/roslyn/pull/34429#discussion_r269208962) [](ancestors = 269208962)